src/control/server/storage/bdev_test.go

```diff
@@ -185,35 +194,73 @@ func Test_NvmeController_Update(t *testing.T) {
 	test.AssertEqual(t, len(mockCtrlrs), 7, "expected 7")
 }
 
-func Test_filterBdevScanResponse(t *testing.T) {
-	const (
-		vmdAddr1         = "0000:5d:05.5"
-		vmdBackingAddr1a = "5d0505:01:00.0"
-		vmdBackingAddr1b = "5d0505:03:00.0"
-		vmdAddr2         = "0000:7d:05.5"
-		vmdBackingAddr2a = "7d0505:01:00.0"
-		vmdBackingAddr2b = "7d0505:03:00.0"
-	)
-	ctrlrsFromPCIAddrs := func(addrs ...string) (ncs NvmeControllers) {
-		for _, addr := range addrs {
-			ncs = append(ncs, &NvmeController{PciAddr: addr})
-		}
-		return
+func Test_NvmeController_Addresses(t *testing.T) {
+	for name, tc := range map[string]struct {
+		ctrlrs   NvmeControllers
+		expAddrs []string
+		expErr   error
+	}{
+		"two vmd endpoints with two backing devices": {
+			ctrlrs: ctrlrsFromPCIAddrs(
+				"5d0505:03:00.0",
+				"7d0505:01:00.0",
+				"5d0505:01:00.0",
+				"7d0505:03:00.0",
+			),
+			expAddrs: []string{
+				"5d0505:01:00.0",
+				"5d0505:03:00.0",
+				"7d0505:01:00.0",
+				"7d0505:03:00.0",
+			},
+		},
+		"no addresses": {
+			expAddrs: []string{},
+		},
+		"invalid address": {
+			ctrlrs: ctrlrsFromPCIAddrs("a"),
+			expErr: errors.New("unable to parse"),
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			gotAddrs, gotErr := tc.ctrlrs.Addresses()
+			test.CmpErr(t, tc.expErr, gotErr)
+			if gotErr != nil {
+				return
+			}
+
+			gotAddrStrs := gotAddrs.Strings()
+			t.Logf("ea: %v\n", gotAddrStrs)
+
+			if diff := cmp.Diff(tc.expAddrs, gotAddrStrs); diff != "" {
+				//if diff := cmp.Diff(tc.expAddrs, gotAddrs.Strings()); diff != "" {
```

Comment:
nit - remove commented line

Reply:
Thanks I'll clean that up in a follow-up